ca/server.go

```diff
@@ -565,6 +576,37 @@ func (s *Server) isRunning() bool {
 	return true
 }
 
+// saveAndFilterExternalCAURLS saves a copy of the external CAs to the server, and also returns a list of external CA
+// urls filtered by the desired cert. This function assumes that something else has called s.signingMu.Lock() and
+// will also unlock it
+func (s *Server) saveAndFilterExternalCAURLS(ctx context.Context, desiredCert, defaultCert []byte, apiExternalCAs []*api.ExternalCA) (urls []string) {
+	desiredCert = NormalizePEMs(desiredCert)
+	s.lastSeenExternalCAs = make([]*api.ExternalCA, len(apiExternalCAs))
```

Comment:
Oops, yes, sorry, although I actually just left this and instead of doing `append`, set `s.lastSeenExternalCAs[i] = ...`.  Don't feel strongly or anything, just setting the value seems to make it more clear that we're copying rather than creating something new.